dagmc_stats/dagmc_stats.py

```diff
@@ -297,3 +324,153 @@ def get_coarseness(my_core, meshset, entity_ranges, geom_dim):
+
+    inputs
+    ------
+    vert_i : vertex entity handle
+    tri_vert_data : numpy structured array that stores the triangle and
+    vertex related data
+
+    outputs
+    -------
+    gc : gaussian curvature value of the vertex
+    """
+    vert_entries = tri_vert_data[tri_vert_data['vert'] == vert_i]
+    sum_alpha_angles = sum(vert_entries['angle'])
+    gc = np.abs(2 * np.pi - sum_alpha_angles)
+    return gc
+
+
+def get_lri(vert_i, gc_all, tri_vert_data, my_core):
+    """Get local roughness value of a vertex
+    Reference: https://www.sciencedirect.com/science/article/pii/
+    S0097849312001203
+    Formula 2, 3
+
+    inputs
+    ------
+    vert_i : vertex entity handle
+    gc_all : dictionary in the form of vertex : gaussian curvature
+    value of the vertex
+    tri_vert_data : numpy structured array that stores the triangle
+    and vertex related data
+    my_core : a MOAB Core instance
+
+    outputs
+    -------
+    Lri : local roughness value of the vertex
+    """
+    DIJgc_sum = 0
+    Dii_sum = 0
+    vert_j_list = list(my_core.get_adjacencies(my_core.get_adjacencies(
+        vert_i, 2, op_type=0), 0, op_type=1))
+    vert_j_list.remove(vert_i)
+    for vert_j in vert_j_list:
+        # get tri_ij_list (the list of the two triangles connected to both
+        # vert_i and vert_j)
+        tri_i_list = my_core.get_adjacencies(vert_i, 2, op_type=0)
+        tri_j_list = my_core.get_adjacencies(vert_j, 2, op_type=0)
+        tri_ij_list = list(set(tri_i_list) & set(tri_j_list))
+        # rows with tri value as tri_ij_list[0] or rei_ij_list[1]
+        select_tris = (tri_vert_data['tri'] == tri_ij_list[0]) | \
+                                    (tri_vert_data['tri'] == tri_ij_list[1])
+        # rows with vert value not equal to vert_i and not equal to vert_j
+        exclude_verts = (tri_vert_data['vert'] != vert_i) & \
+                                            (tri_vert_data['vert'] != vert_j)
+        beta_angles = tri_vert_data[ select_tris & exclude_verts ]['angle']
+
+        Dij = 0.5 * (1/np.tan(beta_angles[0]) + 1/np.tan(beta_angles[1]))
+        DIJgc_sum += (Dij * gc_all[vert_j])
+        Dii_sum += Dij
+    Lri = abs(gc_all[vert_i] - DIJgc_sum/Dii_sum)
+    return Lri
+
+
+def get_roughness(my_core, native_ranges):
+    """Get local roughness values of all the non-isolated vertices
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    native_ranges : a dictionary containing ranges for each native type
+    in the file (VERTEX, TRIANGLE, ENTITYSET)
+
+    outputs
+    -------
+    roughness : (list) the roughness for all surfaces in the meshset
+    """
+    tri_vert_data, all_verts = get_tri_vert_data(my_core,
+                                                    native_ranges[types.MBTRI])
+    gc_all = get_gaussian_curvature(my_core, all_verts, tri_vert_data)
+    roughness = []
+    for vert_i in all_verts:
+        roughness.append(get_lri(vert_i, gc_all, tri_vert_data, my_core))
```

Comment:
after the above change...
```suggestion
    for vert_idx, vert_i in enumerate(all_verts):
        roughness[vert_idx] = get_lri(vert_i, gc_all, tri_vert_data, my_core)
```

Reply:
Thank you very much! By incrementing the index each time instead of flattening(j*3+i as shown in the screenshot) it now works. The order in which the tri_vert data is added affects other parts of the code.

Reply:
I'm happy that this works now @yqin43 , but I'm not sure the order matters.  This will be something that is good to confirm and note if it is true, though!